### PR TITLE
Brian/browser changes

### PR DESCRIPTION
**Pull request by Geometer1729**
other changes
Adds a resume button to the browser app, fixes a bug where the displayed count is not the actual datum.

- [ ] The impure tests via `nix run .#offchain:test` were run and all are passing.
- [x] Where applicable, grammar, punctuation, and spelling within code and comments should be plausibly correct.
- [x] The issue(s) that the work is addressing should be linked in the PR description.
- [x] Any significant changes beyond what's clear from the linked issue(s) are documented in the PR description

### CODE REVIEW

File: offchain/hello-world-browser/src/Main.purs [hunk 1]
```diff
@@ -83,7 +83,7 @@ main =
           , networkId = networkId
           }
       false -> do
-        pure $ testnetNamiConfig { logLevel = Warn }
+        pure $ testnetNamiConfig { logLevel = Debug }
```
Comment (leomayleomay) [inline offchain/hello-world-browser/src/Main.purs]: NITPICK: is this for development purpose? should we change it back to `Warn`?

File: offchain/hello-world-browser/src/Page/Home.purs [hunk 2]
```diff
@@ -39,11 +42,16 @@ _doIncrement
   -> FundsLocked
   -> H.HalogenM State Action slots o m Unit
 _doIncrement datum fundsLocked = do
-  H.modify_ $ const (Incrementing datum (datum + 2))
+  let HelloWorldIncrement inc = helloWorldIncrement
+  H.modify_ $ const (Incrementing datum (datum + inc))
   result <- increment helloWorldIncrement
+  new <- getDatum
```
Comment (leomayleomay) [inline offchain/hello-world-browser/src/Page/Home.purs]: should this be there only if `result` is a `Right`?

File: offchain/hello-world-browser/src/Page/Home.purs [hunk 3]
```diff
@@ -85,12 +93,30 @@ component =
 
   handleAction :: forall slots. Action -> H.HalogenM State Action slots o m Unit
   handleAction = case _ of
+    Resume -> do
+      _ <- H.modify $ const Resuming
```
Comment (leomayleomay) [inline offchain/hello-world-browser/src/Page/Home.purs]:
```suggestion
      H.modify_ $ const Resuming
```

File: offchain/hello-world-browser/src/Page/Home.purs [hunk 4]
```diff
@@ -85,12 +93,30 @@ component =
...
+            Left err -> H.modify_ $ const $ (ResumeFailed err)
+            Right new' -> H.modify_ $ const (Locked new' funds)
+      -- TODO actually query funds locked
+      pure unit
     Lock -> do
       H.modify_ $ const Locking
-      result <- lock helloWorldIncrement 1
+      let init = 1
+      result <- lock helloWorldIncrement init
+      new <- getDatum
```
Comment (leomayleomay) [inline offchain/hello-world-browser/src/Page/Home.purs]: we might want to run this only if `result` is a `Right`

File: offchain/hello-world-browser/src/Page/Home.purs (hunk 2 above)
Comment (Geometer1729) [reply offchain/hello-world-browser/src/Page/Home.purs]: Yup, thanks for catching that :)

**Comment by Geometer1729** [comment]
@epiro-ps  This repo doesn't seem to have the warning when your commit is not current with master, can you look at the pr settings and get them synced with dUSD. I guess that probably goes for DanaSwap too.

File: offchain/hello-world-browser/src/Main.purs (hunk 1 above)
Comment (Geometer1729) [reply offchain/hello-world-browser/src/Main.purs]: Yup, my bad, thanks for catching that

File: offchain/hello-world-browser/src/Page/Home.purs [hunk 5]
```diff
@@ -85,12 +92,29 @@ component =
...
+      H.modify_ $ const Resuming
+      result <- resume helloWorldIncrement
+      case result of
+        Left err -> do
+          H.modify_ $ const (LockFailed err)
+        Right funds -> do
+          new <- getDatum
+          case new of
+            Left err -> H.modify_ $ const $ (ResumeFailed err)
+            Right new' -> H.modify_ $ const (Locked new' funds)
```
Comment (leomayleomay) [inline offchain/hello-world-browser/src/Page/Home.purs]: NITPICK :)

```suggestion
          getDatum >>= case _ of
            Left err -> H.modify_ $ const (ResumeFailed err)
            Right new -> H.modify_ $ const (Locked new funds)
```

Comment (Geometer1729) [reply offchain/hello-world-browser/src/Page/Home.purs]: Thanks for reviewing so carefully. Can't believe I just forgot how to \case between the two fixes :facepalm:.

**Review by leomayleomay — APPROVED** [review]
LGTM

**Review by epiro-ps — APPROVED** [review]

**Pull request merged**